### PR TITLE
Share installation configuration structures

### PR DESCRIPTION
**Pull request by davidcassany**
This commit makes sure elemental-operator and elemental-installer are
using the same structures to describe and define the elemental-cli
installation.

Related to #7

Signed-off-by: David Cassany <dcassany@suse.com>

### CODE REVIEW

**Comment by davidcassany** [comment]
This commit turns `elemental-operator` and `elemental-installer` installation config into a yaml with this syntax:

```yaml
elemental:

  registration:
    url: string
    ca_cert: string
    emulateTPM: bool
    emulatedTPMSeed: int
    noSMBIOS: bool
    labels: map[string]string

  system_agent:
    url: string
    token: string
    secret_name: string
    secret_namespace: string
    
  install:
    firmware: string
    device: string
    no-format: string
    automatic: bool
    config-url: bool
    iso: string
    system-uri: string
    debug: bool
    tty: string
    poweroff: bool
    reboot: bool
    eject-cd: bool
    password: string
    ssh-keys: []string
```

Note variable naming seams to miss any criteria (`no-format`, `secret_name`, `emulatedTPM`, etc.). I'd like to address it, but I guess it deserves another devoted PR once we have some integration tests in place, I don't want to touch it further since it is had to envision the impact of any change.

**Review by kkaempf — CHANGES_REQUESTED** [review]
LGTM, but we should please the linter :wink:

**Comment by davidcassany** [comment]
> LGTM, but we should please the linter wink

Sure, most of them, if not all, are about to be fixed in #18

File: pkg/config/read.go [hunk 1]
```diff
@@ -69,15 +71,11 @@ func ToEnv(cfg Config) ([]string, error) {
...
-	"ELEMENTAL_INSTALL_CONFIG_URL":       "ELEMENTAL_INSTALL_CLOUD_INIT",
-	"ELEMENTAL_INSTALL_ISO_URL":          "ELEMENTAL_INSTALL_ISO",
-	"ELEMENTAL_INSTALL_POWER_OFF":        "ELEMENTAL_POWEROFF",
-	"ELEMENTAL_INSTALL_DEVICE":           "ELEMENTAL_INSTALL_TARGET",
-	"ELEMENTAL_INSTALL_CONTAINER_IMAGE":  "ELEMENTAL_INSTALL_SYSTEM",
-	"ELEMENTAL_INSTALL_NO_FORMAT":        "ELEMENTAL_INSTALL_NO_FORMAT",
-	"ELEMENTAL_INSTALL_TTY":              "ELEMENTAL_INSTALL_TTY",
-	"ELEMENTAL_INSTALL_DEBUG":            "ELEMENTAL_DEBUG",
-	"ELEMENTAL_INSTALL_REGISTRATION_URL": "ELEMENTAL_REGISTRATION_URL",
+	"ELEMENTAL_INSTALL_CONFIG_URL": "ELEMENTAL_INSTALL_CLOUD_INIT",
```
Comment (Itxaka) [inline pkg/config/read.go:74]: isnt there a way we can drop this already? It hurts my heart everytime I see this, it should not eb needed, we control both projects :crying_cat_face:

**Review by Itxaka — APPROVED** [review]

**Pull request merged**